third_party/blink/renderer/core/loader/frame_fetch_context.cc

```diff
@@ -356,6 +356,92 @@ uint64_t RecordReplayNetworkRequestId(uint64_t inspector_id) {
   return recordreplay::RecordReplayValue("NetworkRequestId", inspector_id);
 }
 
+
+static const char* GetRequestCauseString(ResourceRequest& req) {
+  switch (req.GetRequestContext()) {
+    case mojom::blink::RequestContextType::SCRIPT:
+      return "script";
+    case mojom::blink::RequestContextType::STYLE:
+      return "stylesheet";
+    case mojom::blink::RequestContextType::OBJECT:
+      return "object";
+    case mojom::blink::RequestContextType::IMAGE:
+      return "img";
+    case mojom::blink::RequestContextType::XML_HTTP_REQUEST:
+      return "xhr";
+    case mojom::blink::RequestContextType::BEACON:
+      return "beacon";
+    case mojom::blink::RequestContextType::FETCH:
+      return "fetch";
+    case mojom::blink::RequestContextType::XSLT:
+      return "xslt";
+    case mojom::blink::RequestContextType::MANIFEST:
+      return "webManifest";
+    case mojom::blink::RequestContextType::FONT:
+      return "font";
+    case mojom::blink::RequestContextType::PING:
+      return "ping";
+    case mojom::blink::RequestContextType::IMAGE_SET:
+      return "imageset";
+    case mojom::blink::RequestContextType::CSP_REPORT:
+      return "csp";
+    // ReplayIO/Kannan
+    // Remaining are guesses, not quite sure if they're correct (kv).
+    case mojom::blink::RequestContextType::IFRAME:
+    case mojom::blink::RequestContextType::FRAME:
+      return "subdocument";
+    case mojom::blink::RequestContextType::HYPERLINK:
+    case mojom::blink::RequestContextType::PREFETCH:
+      return "document";
+    case mojom::blink::RequestContextType::SUBRESOURCE:
+      return "subdocument";
+    case mojom::blink::RequestContextType::VIDEO:
+      return "media";
+    case mojom::blink::RequestContextType::UNSPECIFIED:
+      return "other";
```

Comment:
Probably best to leave `requestCause` undefined for this too?

Reply:
From the looks of it, it does not matter.
Gecko had `other` and `invalid`, but I agree, here it does not necessarily make sense.

The frontend categorizes it all as ` OTHER` either way (for now):

* https://github.com/replayio/devtools/blob/4c41685584040d762fe5d6dd5a3132f545737d9d/src/ui/components/NetworkMonitor/utils.ts#L182
* https://github.com/replayio/devtools/blob/4c41685584040d762fe5d6dd5a3132f545737d9d/src/ui/components/NetworkMonitor/utils.ts#L79